### PR TITLE
Convert the rest of EtcdClient to util::Task.

### PR DESCRIPTION
**Pull request by pphaneuf**

### CODE REVIEW

**Comment by pphaneuf** [comment]
The latency was killing me, so here we are. I tacked on the next two commits that complete this particular theme... You can look at the separate commits, if you want to get the step-by-step.

File: cpp/util/etcd.cc [hunk 1]
```diff
@@ -499,15 +487,21 @@ EtcdClient::WatchUpdate::WatchUpdate()
...
 
 
-void EtcdClient::WatchInitialGetDone(WatchState* state, util::Status status,
-                                     const Node& node, int64_t etcd_index) {
-  WatchInitialGetAllDone(state, status, {node}, etcd_index);
+void EtcdClient::WatchInitialGetDone(WatchState* state, GetResponse* resp,
+                                     Task* task) {
+  unique_ptr<GetResponse> resp_deleter(resp);
+  GetAllResponse* const all_resp(new GetAllResponse);
+  all_resp->etcd_index = resp->etcd_index;
```
Comment (AlCutter) [inline cpp/util/etcd.cc]: An object `new`/`delete` effectively just to pass in an int, but I guess it's pretty infrequent.

**Comment by AlCutter** [comment]
Nicely Nicelington.

**Comment by AlCutter** [comment]
*Yoink*

**Pull request merged**